src/functionaltests/java/com/ericsson/gerrit/plugins/eiffel/linking/GerritMock.java

```diff
@@ -0,0 +1,129 @@
+        String lookupKey = getLookupKey(user, branch);
+        String changeId = "change-id-" + changeIdsLookup.size();
+
+        assert !changeIdsLookup.containsKey(lookupKey) : lookupKey
+                + " already exists. Forgot to submit it?";
+        changeIdsLookup.put(lookupKey, changeId);
+        ChangeInfo changeInfo = new ChangeInfo();
+        changeInfo.user = user;
+        changeInfo.branch = branch;
+        changeInfos.put(changeId, changeInfo);
+
+        GitCommit branchHead = getHead(branch);
+        GitCommit changeCommit = getNewCommit(branchHead);
+        commits.put(changeId, changeCommit);
+        return changeId;
+    }
+
+    public GitCommit createNewPatchSet(String changeId) {
+        GitCommit oldCommit = commits.get(changeId);
+        GitCommit newCommit = getNewCommit();
+
+        // As we have not rebased the change the startig point has not changed.
+        newCommit.parentSha = oldCommit.parentSha;
+        commits.put(changeId, newCommit);
+        return newCommit;
+    }
+
+    public String getChangeId(String user, String branch) {
+        String lookupKey = getLookupKey(user, branch);
+        return changeIdsLookup.get(lookupKey);
+    }
+
+    public GitCommit rebase(String changeId) {
+        String branch = changeInfos.get(changeId).branch;
+        GitCommit head = getHead(branch);
+        GitCommit rebased = getNewCommit(head);
+        commits.put(changeId, rebased);
+        return rebased;
+
+    }
+
+    public GitCommit submit(String changeId) {
+        GitCommit submittedCommit = getCommit(changeId);
+        String branch = changeInfos.get(changeId).branch;
+
+        // Ensure its rebased
+        assert submittedCommit.parentSha.equals(getHead(branch).sha) : changeId
+                + " is not ready for submit. Reabase it first!";
+
+        branches.put(branch, submittedCommit);
+
+        ChangeInfo changeInfo = changeInfos.get(changeId);
+        String lookupKey = getLookupKey(changeInfo.user, changeInfo.branch);
+        changeIdsLookup.remove(lookupKey);
+        return submittedCommit;
+    }
+
+    public void setExpectionsFor(CommitInformation commitInformation, String changeId,
+            String projectName)
+            throws ResourceNotFoundException, IOException {
+        GitCommit commit = getCommit(changeId);
+        when(commitInformation.getParentsSHAs(commit.sha, projectName)).thenReturn(
+                Arrays.asList(commit.parentSha));
+    }
+
+    private GitCommit getNewCommit() {
+        GitCommit commit = new GitCommit();
+        commit.sha = UUID.randomUUID().toString();
+        return commit;
+    }
+
+    private GitCommit getNewCommit(GitCommit parentCommit) {
+        GitCommit commit = getNewCommit();
+        commit.parentSha = parentCommit.sha;
+        return commit;
+    }
+
+    private String getLookupKey(String user, String branch) {
+        return user + branch;
+    }
+
+    class ChangeInfo {
```

Comment:
Please specify if ChangeInfo is private or public.

Reply:
Fixed in 8d44d79